addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -475,6 +476,8 @@ private List<Transfer> generateNPMContentsFromTransfer( final Transfer transfer,
         try
         {
             ObjectMapper mapper = new ObjectMapper();
+            // Enlarge the max length for a single string value inside of the json
+            mapper.getFactory().setStreamReadConstraints( StreamReadConstraints.builder().maxStringLength(100000000).build() );
```

Comment:
Maybe we can use Integer.MAX_VALUE to avoid this magic number.

Reply:
I think it should be somehow limited to a number based on available memory in the system/pod, but both this or Integer.MAX_VALUE will do for now. Maybe making it configurable would be best.
But more important is probably the behaviour when the limit is reached. IMO it should fail instead of returning 201 as if everything is ok.

Reply:
Um... agree to have a check on the limitation reaching instead of a no-limit setting. @yma96  Will this throw an exception or similar when the limitation is reached?

Reply:
@ligangty, @pkocandr I just updated to the max value of Integer, we could mark it to be configurable for future enhancement.

And for the response, https://github.com/Commonjava/indy/blob/master/addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java#L526, I see it's just recorded as ERROR log output for such IOException, maybe could throw IndyWorkflowException, it seems that broken JSON will result failed or uncompleted uploads, and could let it be noticed. Is there any other concern for this originally? @ligangty, @sswguo

Reply:
@ligangty, It seems that if the JSON size exceeds the maximum StringLength, Jackson will throw StreamConstraintsException. Not sure if that's your concern.

Reply:
@yma96 I don't remember the details of this. I guess this Exception skipping is used to prevent the whole parsing failure from parts of JsonNode parsing error, but I'm also not sure if this is a good idea, as some of the node are key ones, which should fail the whole parsing if they meet parsing error. Is there a way to just recognize this specific node parsing error?

Reply:
> @ligangty, It seems that if the JSON size exceeds the maximum StringLength, Jackson will throw StreamConstraintsException. Not sure if that's your concern.

Sounds like this. Maybe we can catch this exception and fail the upload there with some error response (400 or  something?)

Reply:
@ligangty Yeah, could just catch StreamConstraintsException(Inherited IOException) for the case and do a specific response.